testsupport/signup_request.go

```diff
@@ -0,0 +1,271 @@
+	userIdentity := &authsupport.Identity{
+		ID:       uuid.Must(uuid.NewV4()),
+		Username: r.username,
+	}
+
+	emailClaim0 := authsupport.WithEmailClaim(r.email)
+	token0, err := authsupport.GenerateSignedE2ETestToken(*userIdentity, emailClaim0)
+	require.NoError(r.t, err)
+
+	// Call the signup endpoint
+	invokeEndpoint(r.t, "POST", r.hostAwait.RegistrationServiceURL+"/api/v1/signup",
+		token0, "", r.requiredHTTPStatus)
+
+	// Wait for the UserSignup to be created
+	userSignup, err := r.hostAwait.WaitForUserSignup(userIdentity.ID.String())
+	require.NoError(r.t, err)
+
+	if r.targetCluster != nil {
+		if r.hostAwait.GetToolchainConfig().Spec.Host.AutomaticApproval.Enabled != nil {
+			require.False(r.t, *r.hostAwait.GetToolchainConfig().Spec.Host.AutomaticApproval.Enabled,
+				"cannot specify a target cluster for new signup requests while automatic approval is enabled")
+		}
+	}
+
+	if r.manuallyApprove || r.targetCluster != nil || (r.verificationRequired != states.VerificationRequired(userSignup)) {
+		doUpdate := func(instance *toolchainv1alpha1.UserSignup) {
+			// We set the VerificationRequired state first, because if manuallyApprove is also set then it will
+			// reset the VerificationRequired state to false.
+			if r.verificationRequired != states.VerificationRequired(instance) {
+				states.SetVerificationRequired(userSignup, r.verificationRequired)
+			}
+
+			if r.manuallyApprove {
+				states.SetApproved(instance, r.manuallyApprove)
+			}
+			if r.targetCluster != nil {
+				instance.Spec.TargetCluster = r.targetCluster.ClusterName
+			}
+		}
+
+		userSignup, err = r.hostAwait.UpdateUserSignupSpec(userSignup.Name, doUpdate)
+		require.NoError(r.t, err)
+	}
+
+	r.t.Logf("user signup '%s' created", userSignup.Name)
+
+	// If any required conditions have been specified, confirm the UserSignup has them
+	if len(r.conditions) > 0 {
+		userSignup, err = r.hostAwait.WaitForUserSignup(userSignup.Name, wait.UntilUserSignupHasConditions(r.conditions...))
+		require.NoError(r.t, err)
+	}
+
+	r.userSignup = userSignup
+
+	if r.ensureMUR {
+		// Confirm the MUR was created and ready
+		VerifyResourcesProvisionedForSignup(r.t, r.hostAwait, userSignup, "base", r.memberAwait, r.member2Await)
+		mur, err := r.hostAwait.WaitForMasterUserRecord(userSignup.Status.CompliantUsername)
+		require.NoError(r.t, err)
+		r.mur = mur
+	}
+
+	// We also need to ensure that the UserSignup is deleted at the end of the test (if the test itself doesn't delete it)
+	r.t.Cleanup(func() {
+		propagationPolicy := metav1.DeletePropagationForeground
+		opts := client.DeleteOption(&client.DeleteOptions{
+			PropagationPolicy: &propagationPolicy,
+		})
+
+		// Delete the UserSignup
+		if err := r.hostAwait.Client.Delete(context.TODO(), r.userSignup, opts); err != nil && !errors.IsNotFound(err) {
+			require.NoError(r.t, err)
+		}
+
+		// Ensure the UserSignup has been deleted
+		require.NoError(r.t, r.hostAwait.WaitUntilUserSignupDeleted(r.userSignup.Name))
+	})
```

Comment:
@sbryzak You forgot to remove this code and replace it with the `Awaitility.Cleanup` function. Could you please fix that?